trick_sims/SIM_wheelbot/models/Control/src/differentialDriveController.cpp

```diff
@@ -23,7 +23,14 @@ DifferentialDriveController::
     leftMotorController(LeftMotorController),
 
     rightMotorSpeedCommand(0.0),
-    leftMotorSpeedCommand(0.0)
+    leftMotorSpeedCommand(0.0),
+    desiredHeadingRate(0.0),
+
+    // PID Controller Initialization
+    headingctrl(new PIDController(1.0, 0.08, 0.5, headingRateLimit,
```

Comment:
This class needs a destructor if you are going to be allocating memory with 'new' in the constructor. Alternatively, you can change the pointers to regular objects which would remove the need for a destructor, assignment operator, copy constructor. From a glance I do not see any reason that you need to use pointers instead of objects.